db_test.go

```diff
@@ -641,3 +644,116 @@ func TestDB_ConcurrentMapWrite(t *testing.T) {
+	replicaPath := filepath.Join(t.TempDir(), "replica")
+	client := file.NewReplicaClient(replicaPath)
+	db.Replica = litestream.NewReplicaWithClient(db, client)
+	db.Replica.MonitorEnabled = false
+
+	// Create some transactions
+	for i := 0; i < 10; i++ {
+		if _, err := sqldb.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS t (id INTEGER PRIMARY KEY, val TEXT)`); err != nil {
+			t.Fatalf("create table: %v", err)
+		}
+		if _, err := sqldb.ExecContext(ctx, `INSERT INTO t (val) VALUES (?)`, fmt.Sprintf("value-%d", i)); err != nil {
+			t.Fatalf("insert %d: %v", i, err)
+		}
+
+		// Sync to create L0 files
+		if err := db.Sync(ctx); err != nil {
+			t.Fatalf("sync db: %v", err)
+		}
+		if err := db.Replica.Sync(ctx); err != nil {
+			t.Fatalf("sync replica: %v", err)
+		}
+	}
+
+	// Record the earliest L0 file timestamp before compaction
+	itr, err := client.LTXFiles(ctx, 0, 0, time.Time{})
+	if err != nil {
+		t.Fatalf("list L0 files: %v", err)
+	}
+	defer itr.Close()
+
+	l0Files, err := ltx.SliceFileIterator(itr)
+	if err != nil {
+		t.Fatalf("convert iterator: %v", err)
+	}
+	if err := itr.Close(); err != nil {
+		t.Fatalf("close iterator: %v", err)
+	}
+
+	var earliestTime time.Time
+	for _, info := range l0Files {
+		if earliestTime.IsZero() || info.CreatedAt.Before(earliestTime) {
+			earliestTime = info.CreatedAt
+		}
+	}
+
+	if len(l0Files) == 0 {
+		t.Fatal("expected L0 files before compaction")
+	}
+	t.Logf("Found %d L0 files, earliest timestamp: %v", len(l0Files), earliestTime)
+
+	// Perform compaction from L0 to L1
+	levels := litestream.CompactionLevels{
+		{Level: 0},
+		{Level: 1, Interval: time.Second},
+	}
+	store := litestream.NewStore([]*litestream.DB{db}, levels)
+	store.CompactionMonitorEnabled = false
+
+	if err := store.Open(ctx); err != nil {
+		t.Fatalf("open store: %v", err)
+	}
+	defer func() {
+		if err := store.Close(ctx); err != nil {
+			t.Fatalf("close store: %v", err)
+		}
+	}()
+
+	_, err = store.CompactDB(ctx, db, levels[1])
+	if err != nil {
+		t.Fatalf("compact: %v", err)
+	}
+
+	// Verify L1 file has the earliest timestamp from L0 files
+	itr, err = client.LTXFiles(ctx, 1, 0, time.Time{})
+	if err != nil {
+		t.Fatalf("list L1 files: %v", err)
+	}
+	defer itr.Close()
+
+	l1Files, err := ltx.SliceFileIterator(itr)
+	if err != nil {
+		t.Fatalf("convert L1 iterator: %v", err)
+	}
+	if err := itr.Close(); err != nil {
+		t.Fatalf("close L1 iterator: %v", err)
+	}
+
+	if len(l1Files) == 0 {
+		t.Fatal("expected L1 file after compaction")
+	}
+
+	l1Info := l1Files[0]
+
+	// The L1 file's CreatedAt should be the earliest timestamp from the L0 files
+	// Allow for some drift due to millisecond precision in LTX headers
+	timeDiff := l1Info.CreatedAt.Sub(earliestTime)
+	if timeDiff.Abs() > time.Second {
+		t.Errorf("L1 CreatedAt = %v, earliest L0 = %v (diff: %v)", l1Info.CreatedAt, earliestTime, timeDiff)
+		t.Error("L1 file timestamp should preserve earliest source file timestamp")
+	}
```

Comment:
I'm surprised it is using the earliest timestamp. The `ltx.Compactor` uses the last timestamp from the input files here: https://github.com/superfly/ltx/blob/2e6df57fc041819c837bba4f94438fec5868b85e/compactor.go#L85

The idea is that once you compact all those files together you get the state of the database of the last transaction in that file group so you'd want the last timestamp.